pages/en/android/camera-orientation.md

```diff
@@ -0,0 +1,301 @@
+
+![Image illustration of a portrait-shaped preview stretched to fit inside a TextureView of the same size of the preview size chosen](/images/android/camera-orientation/texture_view_step_1_identical_size.png)
+
+##### 2. Scale the potentially stretched TextureView back to the original dimensions of the preview
+
+Consider the following to scale the stretched preview back to the dimensions of the source.
+
+Dimensions of the source (`sourceWidth × sourceHeight`) is:
+
+- `previewHeight × previewWidth`, if the natural orientation is portrait or reverse-portrait (sensor orientation is 90 or 270 degrees)
+- `previewWidth × previewHeight`, if the natural orientation is landscape or reverse-landscape (sensor orientation is 0 or 180 degrees)
+
+Fix stretching by utilizing [View#setScaleX(float)](<https://developer.android.com/reference/android/view/View#setScaleX(float)>) and [View#setScaleY(float)](<https://developer.android.com/reference/android/view/View#setScaleY(float)>)
+
+- setScaleX(`sourceWidth / previewWidth`)
+- setScaleY(`sourceHeight / previewHeight`)
+
+![Image illustration showing the procedure of the stretched preview being scaled back to its original dimensions](/images/android/camera-orientation/texture_view_step_2_scale_back.png)
+
+##### 3. Rotate the preview by `displayRotation` counterclockwise
+
+As previously mentioned, you should rotate the preview by `displayRotation` counterclockwise to compensate for display rotation.
+
+You can do this by [View#setRotation(float)](<https://developer.android.com/reference/android/view/View#setRotation(float)>)
+
+- setRotation(`-displayRotation`), since it does a clockwise rotation.
+
+![Image illustration showing the procedure of the preview being rotated to match to the display orientation of the device](/images/android/camera-orientation/texture_view_step_3_rotate.png)
+
+##### Sample
+
+- [PreviewView](https://developer.android.com/reference/androidx/camera/view/PreviewView) from [camerax](https://developer.android.com/training/camerax) in [Jetpack](https://developer.android.com/android/jetpack) handles TextureView layout as described above. It configures the transformation with [PreviewCorrector](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-master-dev:camera/camera-view/src/main/java/androidx/camera/view/preview/transform/PreviewCorrector.java).
+
+##### Note
+
+If you have previously used transformation matrix for TextureView in your code, the preview may not look right on a naturally-landscape device like Chromebooks. Likely your transformation matrix incorrectly assumes the sensor orientation to be 90 or 270 degrees. You may refer to this [commit](https://github.com/android/camera-samples/commit/3d1a254eb018a51ff39ae78d39a9e9e7942a027b) on GitHub for a workaround, but we highly recommend that you migrate your app to use the method described here instead.
```

Comment:
```suggestion
!!! aside.message--note
**Note:** If you have previously used transformation matrix for TextureView in your code, the preview may not look right on a naturally-landscape device like Chromebooks. Likely your transformation matrix incorrectly assumes the sensor orientation to be 90 or 270 degrees. You may refer to this [commit](https://github.com/android/camera-samples/commit/3d1a254eb018a51ff39ae78d39a9e9e7942a027b) on GitHub for a workaround, but we highly recommend that you migrate your app to use the method described here instead.
!!!
```

We have a specific set of markdown fences to render this content as a note that should be used here.